front/components/app/blocks/Input.tsx

```diff
@@ -40,12 +46,62 @@ export default function Input({
   onBlockDown: () => void;
   onBlockNew: (blockType: BlockType | "map_reduce" | "while_end") => void;
 }>) {
-  const handleSetDataset = (dataset: string) => {
+  const [isDatasetModalOpen, setIsDatasetModalOpen] = useState(false);
+  const { dataset, isDatasetLoading, isDatasetError } = useDataset(
+    owner,
+    app,
+    block.config.dataset,
+    true
+  );
+
+  const [datasetModalData, setDatasetModalData] = useState<DatasetType | null>(
+    null
+  );
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const handleSetDataset = async (datasetName: string) => {
```

Comment:
can we keep `dataset`? this is indeed the name but this is also an `id` and using `datasetName` really does not convey that :pray: